weaver/cli.py

```diff
@@ -0,0 +1,747 @@
+            result = self.undeploy(process_id=process_id, url=base)
+            if result.code not in [200, 404]:
+                return OperationResult(False, "Failed requested undeployment prior deployment.",
+                                       body=result.body, text=result.text, code=result.code, headers=result.headers)
+        path = f"{base}/processes"
+        resp = request_extra("POST", path, json=data, headers=headers, settings=self._settings)
+        return self._parse_result(resp)
+
+    def undeploy(self, process_id, url=None):
+        # type: (str, Optional[str]) -> OperationResult
+        """
+        Undeploy an existing :term:`Process`.
+
+        :param process_id: Identifier of the process to undeploy.
+        :param url: Instance URL if not already provided during client creation.
+        """
+        base = self._get_url(url)
+        path = f"{base}/processes/{process_id}"
+        resp = request_extra("DELETE", path, headers=self._headers, settings=self._settings)
+        return self._parse_result(resp)
+
+    def capabilities(self, url=None):
+        # type: (Optional[str]) -> OperationResult
+        """
+        List all available :term:`Process` on the instance.
+
+        :param url: Instance URL if not already provided during client creation.
+        """
+        base = self._get_url(url)
+        path = f"{base}/processes"
+        query = {"detail": False}  # not supported by non-Weaver, but save the work if possible
+        resp = request_extra("GET", path, params=query, headers=self._headers, settings=self._settings)
+        result = self._parse_result(resp)
+        processes = result.body.get("processes")
+        if isinstance(processes, list) and all(isinstance(proc, dict) for proc in processes):
+            processes = [get_any_id(proc) for proc in processes]
+            result.body = processes
+        return result
+
+    processes = capabilities  # alias
+    """
+    Alias of :meth:`capabilities` for :term:`Process` listing.
+    """
+
+    def describe(self, process_id, url=None):
+        # type: (str, Optional[str]) -> OperationResult
+        """
+        Describe the specified :term:`Process`.
+
+        :param process_id: Identifier of the process to describe.
+        :param url: Instance URL if not already provided during client creation.
+        """
+        base = self._get_url(url)
+        path = f"{base}/processes/{process_id}"
+        resp = request_extra("GET", path, headers=self._headers, settings=self._settings)
+        return self._parse_result(resp)
+
+    # FIXME: support sync (https://github.com/crim-ca/weaver/issues/247)
+    # :param execute_async:
+    #   Execute the process asynchronously (user must call :meth:`monitor` themselves,
+    #   or synchronously where monitoring is done automatically until completion before returning.
+    def execute(self, process_id, inputs=None, monitor=False, timeout=None, interval=None, url=None):
```

Comment:
By default no monitoring is done here. It is a quality of life feature.
The user must request it explicitly (`-M`) if they want it, and optionally with different times (`-T` and `-W`).
This is 100% equivalent to doing `execute` (without `-M`), manually copy-paste the printed job ID, and call `monitor` with it.

Indeed, monitoring would be relevant in `sync` because the response is obtained only after full completion. 
Here, `sync` is not a CLI feature, but rather the actual API execution mode. 
This is not implemented in Weaver for the moment (https://github.com/crim-ca/weaver/issues/247)

Reply:
Hum, I'm still confused. There are 3 scenarios:
1. I want to execute a process sync (which Weaver doesn't support)
2. I want to execute a process async and monitor it from the execute function
3. I want to execute a process async and deal with the monitoring
So the monitor flag control 2 or 3? And the function doesn't handle 1? If this is the case, that's fine. I was thinking that the monitor flag was also setting the async API flag.

Reply:
Yes, that's it.